### PR TITLE
Set up R package structure and testing infrastructure

### PR DESCRIPTION
**Pull request by terrytangyuan**
* Set up R package structure
* Set up testing infrastructure and added a couple test utility functions

### CODE REVIEW

**Comment by terrytangyuan** [comment]
@yongtang It's `devtools::test("path_to_r_package")`. I'll prepare a Docker image and some instructions in a follow-up PR.

File: R-package/DESCRIPTION [hunk 1]
```diff
@@ -0,0 +1,38 @@
+Package: tfio
```
Comment (yongtang) [inline R-package/DESCRIPTION]: Wondering if for cosmetic reasons is it possible to change it to tensorflow-io to match python package?

**Comment by yongtang** [comment]
I tried to install devtools from the default r-base (with DockerHub's r-base (https://hub.docker.com/_/r-base/). After invoking `install.packages("devtools", dependencies = TRUE)`, the dependency fails with:
```
ERROR: dependencies ‘curl’, ‘httr’, ‘rvest’, ‘xml2’ are not available for package ‘foghorn’
* removing ‘/usr/local/lib/R/site-library/foghorn’
ERROR: dependencies ‘httr’, ‘whoami’ are not available for package ‘rhub’
* removing ‘/usr/local/lib/R/site-library/rhub’
ERROR: dependency ‘xml2’ is not available for package ‘roxygen2’
* removing ‘/usr/local/lib/R/site-library/roxygen2’
ERROR: dependencies ‘git2r’, ‘httr’, ‘usethis’ are not available for package ‘devtools’
* removing ‘/usr/local/lib/R/site-library/devtools’
ERROR: dependencies ‘httr’, ‘openssl’, ‘roxygen2’, ‘rsconnect’, ‘xml2’ are not available for package ‘pkgdown’
* removing ‘/usr/local/lib/R/site-library/pkgdown’
```

 is there any other ways of install devtools?

File: R-package/DESCRIPTION (hunk 1 above)
Comment (terrytangyuan) [reply R-package/DESCRIPTION]: R package name cannot contain dashes. Also I was trying to be consistent with other existing R interfaces to TF components:
* `tf.estimator`: https://github.com/rstudio/tfestimators
* `tf.dataset`: https://github.com/rstudio/tfdatasets

**Comment by terrytangyuan** [comment]
@yongtang `ERROR: dependencies ‘curl’, ‘httr’, ‘rvest’, ‘xml2’ are not available` - this indicates that there are several system dependencies needed. Try installing these before installing `devtools`: `sudo apt-get install build-essential libcurl4-gnutls-dev libxml2-dev libssl-dev`.

**Comment by yongtang** [comment]
Thanks! After installed `devtools` and `testthat`, then`tensorflow`, the `devtools::test()` worked from R interface.

**Comment by yongtang** [comment]
It looks like Travis CI also support R. I will take to see if it is possible to invoke R test with Travis CI.

**Comment by terrytangyuan** [comment]
Yes, Travis is very friendly with R. If you encounter any issues, let me know!

**Pull request merged**